### ISSUE TITLE
Docker Compose Updates

### ISSUE THREAD

**Issue by TheBurchLog** [beer-garden/beer-garden#348]
When attempting to deploy via the docker-composer file ran into issues in the following scenario.

Environment:
Deployed Mongodb and RabbitMQ via Docker-Composer
Deployed Bartender and Brew-View via local shell script

Issues:
MongoDB does not expose the ports required
RabbitMQ does not set the default user-name/password to what BeerGarden is expecting.

**Comment by hazmat345** [comment]
Mongo port issue is fixed in 638fb39.

**Comment by hazmat345** [comment]
The rabbit user/password is a little tricker. Ideally things would "just work" - you'd be able to install rabbit, clone the beergarden repos, start everything up *with no configuration changes*, and be good to go.

We run into this issue because we want to support two different ways of installing rabbit:

- An rpm. With no config changes this supports a username / password combo of "guest/guest". Config files in the beergarden repos are currently set to this, so an rpm install of rabbitmq should work right away.
- A docker image using our provided docker-compose file. Here we can specify the user/password combo that rabbit will use. If we could use "guest/guest" here there would be no problem, but unfortunately rabbit does not all logins using "guest" from any url other than "localhost". There's no way to get that working for all possible cases with regards to remote plugins without using "host" networking for everything, and there's obviously no way to make that work with TLS at all. So we set a different user/password in the docker case to allow for "remote" logins.

Bottom line - there's no way to have a standard configuration that works in all cases. Options are:

- Set the user/pass in the beergarden config files to the rabbit rpm defaults (the current way) and accept that you'll need to change them if using a docker rabbit install.
- Set the user/pass in the beergarden config files to the docker defaults, and accept that if using an rpm install you'll either need to
  - change the config files OR
  - Add the "beer_garden" user to the rabbit installation yourself

**Comment by hazmat345** [comment]
@TheBurchLog this isn't really a v2 problem, marking it v3 for triage.

**Comment by hazmat345** [comment]
This needs to be handles in docs, closing.
